responsibleai/responsibleai/_managers/counterfactual_manager.py

```diff
@@ -329,8 +365,88 @@ def name(self):
         return ManagerNames.COUNTERFACTUAL
 
     def _save(self, path):
-        pass
+        """Save the CounterfactualManager to the given path.
 
-    @staticmethod
-    def _load(path, model_analysis):
-        pass
+        :param path: The directory path to save the CounterfactualManager to.
+        :type path: str
+        """
+        counterfactual_dir = Path(path)
+        counterfactual_dir.mkdir(parents=True, exist_ok=True)
+        for counterfactual_config in self._counterfactual_config_list:
+            counterfactual_state_dir = Path(path) / str(uuid.uuid4())
+            counterfactual_state_dir.mkdir(parents=True, exist_ok=True)
+
+            config_path = counterfactual_state_dir / 'config.json'
+            with open(config_path, 'w') as config_file:
+                json.dump(
+                    counterfactual_config.get_config_as_dict(),
+                    config_file)
+
+            result_path = counterfactual_state_dir / 'result.json'
+            with open(result_path, 'w') as result_file:
+                json.dump(
+                    counterfactual_config.get_result(),
+                    result_file)
+
+    @classmethod
+    def _load(cls, path, model_analysis):
+        """Load the CounterfactualManager from the given path.
+
+        :param path: The directory path to load the CounterfactualManager from.
+        :type path: str
+        :param model_analysis: The loaded parent ModelAnalysis.
+        :type model_analysis: ModelAnalysis
```

Comment:
is this missing return type?  return/rtype

Reply:
Done